server/publications/collections/orders.js

```diff
@@ -63,10 +65,51 @@ Meteor.publish("CustomPaginatedOrders", function (query, options) {
   if (!shopId) {
     return this.ready();
   }
+
+  // return any order for which the shopId is attached to an item
+  const selector = {
+    "items.shopId": shopId
+  };
   if (Roles.userIsInRole(this.userId, ["admin", "owner", "orders"], shopId)) {
-    Counts.publish(this, "order-count", Orders.find({ shopId: shopId }), { noReady: true });
-    return Orders.find({ shopId: shopId });
+    ReactiveAggregate(this, Orders, [
+      {
+        $project: {
+          items: {
+            $filter: {
+              input: "$items",
+              as: "item",
+              cond: { $eq: ["$$item.shopId", shopId] }
+            }
+          },
+          billing: {
+            $filter: {
+              input: "$billing",
+              as: "billing",
+              cond: { $eq: ["$$billing.shopId", shopId] }
+            }
+          },
+          shipping: {
+            $filter: {
+              input: "$shipping",
+              as: "shipping",
+              cond: { $eq: ["$$shipping.shopId", shopId] }
+            }
+          },
+          cartId: 1,
+          sessionId: 1,
+          shopId: 1,
+          workflow: 1,
```

Comment:
We probably should make sure that any give shop can't move the overall order workflow forward. The overall workflow should only be progressed/regressed by some type of system process or hook that knows when all shipments/payments/items have been progressed.

Also discussion topic, maybe we don't use this field at all? Maybe we keep it for marketplace?